hack/update-generated-code.sh

```diff
@@ -14,18 +14,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ROOT=$(unset CDPATH && cd $(dirname "${BASH_SOURCE[0]}")/.. && pwd)
-cd $ROOT
-
 set -o errexit
 set -o nounset
 set -o pipefail
 
-source "${ROOT}/hack/lib.sh"
-
-hack::build_generators
+SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
 
-ALL_FQ_APIS=github.com/kubernetes-incubator/external-storage/snapshot/pkg/apis/crd/v1
+# generate the code with:
+# --output-base    because this script should also be able to run inside the vendor dir of
+#                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
+#                  instead of the $GOPATH directly. For normal projects this can be dropped.
+${CODEGEN_PKG}/generate-groups.sh "deepcopy,client,informer,lister" \
+  github.com/kubernetes-incubator/external-storage/snapshot/pkg/client github.com/kubernetes-incubator/external-storage/snapshot/pkg/apis \
+  volumesnapshot:v1 \
+  --go-header-file ${SCRIPT_ROOT}/hack/boilerplate.go.txt
 
-echo "Generating deepcopy funcs"
-${OUTPUT_HOSTBIN}/deepcopy-gen --input-dirs ${ALL_FQ_APIS} -O zz_generated.deepcopy --bounding-dirs $ALL_FQ_APIS --logtostderr
+# To use your own boilerplate text use:
+#   --go-header-file ${SCRIPT_ROOT}/hack/custom-boilerplate.go.txt
```

Comment:
I failed to run hack/update-generated-code.sh successfully. It spits this and ends with error:
```
# github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/informer-gen/args
/home/tsmetana/go/src/github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/informer-gen/args/args.go:38:31: args.Default().WithoutDefaultFlagParsing undefined (type *args.GeneratorArgs has no field or method WithoutDefaultFlagParsing)
# github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/defaulter-gen/args
/home/tsmetana/go/src/github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/defaulter-gen/args/args.go:32:31: args.Default().WithoutDefaultFlagParsing undefined (type *args.GeneratorArgs has no field or method WithoutDefaultFlagParsing)
# github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/client-gen/args
/home/tsmetana/go/src/github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/client-gen/args/args.go:55:31: args.Default().WithoutDefaultFlagParsing undefined (type *args.GeneratorArgs has no field or method WithoutDefaultFlagParsing)
# github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/lister-gen/args
/home/tsmetana/go/src/github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/lister-gen/args/args.go:33:31: args.Default().WithoutDefaultFlagParsing undefined (type *args.GeneratorArgs has no field or method WithoutDefaultFlagParsing)
# github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/deepcopy-gen/args
/home/tsmetana/go/src/github.com/kubernetes-incubator/external-storage/vendor/k8s.io/code-generator/cmd/deepcopy-gen/args/args.go:32:31: args.Default().WithoutDefaultFlagParsing undefined (type *args.GeneratorArgs has no field or method WithoutDefaultFlagParsing)
```
Not sure what exactly is wrong but the code generation definitely works in the master branch... Obviously the resulting code does not work...